### PR TITLE
Fix taglocation waiter

### PR DESCRIPTION
**Pull request by ghabian**
This PR fixes a bug in tags_location_tag_binding resource for zonal parent resources.

For resources that return an operation ID on create/destory. We need to read the location and update the basepath in the created waiters.

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/11448

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
tags: added support for zonal resources in parent for `tags_location_tag_binding` 

```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
Hey, can you help us to identify which resource(s) are affected, and which tests are covering the change? Thanks in advance.

**Comment by ghabian** [comment]
the resource impacted is `tags_location_tag_binding` already has tests written for it. I'm fixing a bug when using the resource in a way not covered by the existing tests.

Should I write tests specifically covering this scenario and add them as part of the PR?

**Comment by hao-nan-li** [comment]
> the resource impacted is `tags_location_tag_binding` already has tests written for it. I'm fixing a bug when using the resource in a way not covered by the existing tests.
> 
> Should I write tests specifically covering this scenario and add them as part of the PR?

Yeah, I think it's a good idea to add tests that cover the scenario you've mentioned.

**Review by hao-nan-li — CHANGES_REQUESTED** [review]
Any update so far? You might also need to do a rebase on the main branch in order to run builds on git in the future.

**Comment by ghabian** [comment]
@hao-nan-li added a test to do zonal bindings which passes in my local build. `testAccTagsLocationTagBinding_locationTagBindingzonal`

File: mmv1/third_party/terraform/tests/resource_tags_test.go [hunk 1]
```diff
@@ -879,3 +880,76 @@ func testAccCheckTagsLocationTagBindingDestroyProducer(t *testing.T) func(s *ter
 		return nil
 	}
 }
+
+
+
+func testAccTagsLocationTagBinding_locationTagBindingzonal(t *testing.T) {
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/tests/resource_tags_test.go:886]: Thanks for adding the tests, can you make the first letter capital case? 
`TestAccTagsLocationTagBinding_locationTagBindingzonal`

**Review by hao-nan-li — CHANGES_REQUESTED** [review]
It looks like the PR has some linting problem. Could you run `go vet` locally to double-check?

When i had the same issue previously, I had to add an extra line at the end the files that has the lint alert.

**Comment by ghabian** [comment]
> It looks like the PR has some linting problem. Could you run `go vet` locally to double-check?
> 
> When i had the same issue previously, I had to add an extra line at the end the files that has the lint alert.

`go vet` shows me nothing even after I sync/pull from the main branch. for both ga and beta providers.

**Review by hao-nan-li — CHANGES_REQUESTED** [review]
Looks like the lint errors are still there. Can you add an extra line at the end of each file that does not have an empty line?

**Review by hao-nan-li — CHANGES_REQUESTED** [review]
Looks like there are more lint problems to work on.

I created a branch with your commits and did the following:

haonan@haonan-macbookpro terraform-provider-google % **make lint**
sh -c "'/Users/haonan/go/src/github.com/hashicorp/terraform-provider-google/scripts/gofmtcheck.sh'"
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
**./google/tagslocation_operation.go
./google/resource_tags_test.go
./google/self_link_helpers.go**
You can use the command: `make fmt` to reformat code.

**Comment by ghabian** [comment]
> Looks like there are more lint problems to work on.
> 
> I created a branch with your commits and did the following:
> 
> haonan@haonan-macbookpro terraform-provider-google % **make lint** sh -c "'/Users/haonan/go/src/github.com/hashicorp/terraform-provider-google/scripts/gofmtcheck.sh'" ==> Checking that code complies with gofmt requirements... gofmt needs running on the following files: **./google/tagslocation_operation.go ./google/resource_tags_test.go ./google/self_link_helpers.go** You can use the command: `make fmt` to reformat code.

thanks, latest commit fixes those

**Review by hao-nan-li — APPROVED** [review]
LGTM, waiting on @zli82016 to sign off.

Just a quick note to change the release message following the release note guide.

**Comment by ghabian** [comment]
> LGTM, waiting on @zli82016 to sign off.
Thanks!

> Just a quick note to change the release message following the release note guide.
Updated, PTAL

File: mmv1/third_party/terraform/utils/self_link_helpers.go [hunk 2]
```diff
@@ -181,3 +181,14 @@ func GetRegionFromRegionalSelfLink(selfLink string) string {
 	}
 	return selfLink
 }
+
+func GetLocationFromOpName(opName string) string {
```
Comment (zli82016) [inline mmv1/third_party/terraform/utils/self_link_helpers.go:185]: Can you please move this function to the file tagslocation_operation.go, as it is only used there for now?

File: mmv1/third_party/terraform/utils/tagslocation_operation.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package google
```
Comment (zli82016) [inline mmv1/third_party/terraform/utils/tagslocation_operation.go:1]: Can you please rename this file to tags_location_operation.go to follow the convention?

File: mmv1/third_party/terraform/utils/tagslocation_operation.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		Config:    config,
+		UserAgent: userAgent,
+	}
+	if err := w.CommonOperationWaiter.SetOp(op); err != nil {
+		return nil, err
+	}
+	return w, nil
+}
+
+// nolint: deadcode,unused
```
Comment (zli82016) [inline mmv1/third_party/terraform/utils/tagslocation_operation.go:42]: Can you please remove this line, as the function tagsLocationOperationWaitTimeWithResponse is used?

**Comment by zli82016** [comment]
@ghabian , any update? Thanks.

**Comment by ghabian** [comment]
> @ghabian , any update? Thanks.

Done, PTAL

**Comment by zli82016** [comment]
Hello, @ghabian, can you please rebase main branch? Recently, ruby is upgraded to 3.2.0.

**Comment by ghabian** [comment]
rebased just now

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**